src/http_server.cc

```diff
@@ -267,6 +268,26 @@ HTTPServer::Dispatch(evhtp_request_t* req, void* arg)
   (static_cast<HTTPServer*>(arg))->Handle(req);
 }
 
+evhtp_res
+HTTPServer::NewConnection(evhtp_connection_t* conn, void* arg)
```

Comment:
does any of this need a fence or mutex?

Reply:
I am not aware of any need from the evhtp library side, the example I found online does not use any fence/lock/mutex: https://chromium.googlesource.com/external/github.com/ellzey/libevhtp/+/refs/heads/master/examples/test.c#349

On our HTTP server side, the only thing shared is the `void* arg` which is simply `this`, the only two variables it interacts on are `accept_new_conn_` and `conn_cnt_` which are already safeguarded by `std::atomic<...>`. Here is their definition:
```
std::atomic<bool> accept_new_conn_;
std::atomic<uint32_t> conn_cnt_;
```

It is ok for them to use separate "mutexes" because the flow on the `HTTPServer::NewConnection()` is to take a peek into `accept_new_conn_`. If it is `false, the new connection is rejected (connection reset). If it is `true`, the function moves on to `conn_cnt_` and does not interact with `accept_new_conn_` anymore.